__tests__/apiPostArticleComment.test.js

```diff
@@ -0,0 +1,79 @@
+const request = require("supertest");
+const app = require("../index");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const data = require("../db/data/test-data");
+const jestSorted = require("jest-sorted");
+
+afterAll(() => {
+  db.end();
+});
+
+beforeEach(() => {
+  return seed(data);
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  it("checking the body of returned comment", async () => {
+    const commentBody = "small cat";
+    const username = "lurker";
+    const articleId = 4;
+
+    const response = await request(app)
+      .post(`/api/articles/${articleId}/comments`)
+      .send({ username: username, body: commentBody });
+
+    expect(response.status).toEqual(201);
+    expect(response.body).toBeInstanceOf(Object);
+
+    expect(response.body).toEqual({ comment: "small cat" });
+  });
+});
+
+it("returns 404 if the article_id is not a number", async () => {
+  const commentBody = "small cat";
+  const username = "lurker";
+  const response = await request(app)
+    .post("/api/articles/notANumber/comment")
+    .send({ username, body: commentBody })
+    .catch((err) => {
+      expect(err.response.status).toBe(400);
+    });
+});
+
+it("expect 404 code when inputting invalid username ", () => {
+  return request(app)
+    .post("/api/articles/3/comments")
+    .send({ username: "Duncan", body: "small cat" })
+    .catch((err) => {
+        expect(err.response.status).toBe(404);
+      });
```

Comment:
This is promise based, but the others you've used async/await - id stick to one

Reply:
Desc says invalid - which suggests 400, but the error code is 404 - description needs to be something about username not found

Reply:
Done